src/index.js

```diff
@@ -1,21 +1,44 @@
 import 'isomorphic-fetch';
+import promiseRetry from 'promise-retry';
 import objectAssign from 'object-assign';
 
-export default function jsonFetch(URL, fetchOptions = {}) {
-  let fetchOptionsClone = objectAssign({}, fetchOptions);
+export {retriers} from './retriers';
+
+const DEFAULT_RETRY_OPTIONS = {
+  retries: 0,
+  shouldRetry: () => false
+};
+
+export default function jsonFetch(URL, options = {}) {
+  let fetchOptions = objectAssign({}, options);
+  let retryOptions = objectAssign(DEFAULT_RETRY_OPTIONS, options.retry);
   const jsonHeaders = {
     'Accept': 'application/json',
     'Content-Type': 'application/json',
   };
 
-  if (fetchOptionsClone.credentials === undefined) fetchOptionsClone.credentials = 'include';
+  if (fetchOptions.credentials === undefined) fetchOptions.credentials = 'include';
 
-  fetchOptionsClone.headers = objectAssign({}, jsonHeaders, fetchOptions.headers);
+  fetchOptions.headers = objectAssign({}, jsonHeaders, fetchOptions.headers);
   try {
-    fetchOptionsClone.body = JSON.stringify(fetchOptions.body);
+    fetchOptions.body = JSON.stringify(fetchOptions.body);
   } catch (err) {}
 
-  return fetch(URL, fetchOptionsClone)
+  return promiseRetry((retry) => {
+    return fetch(URL, fetchOptions)
+      .then((response) => {
+        if(retryOptions.shouldRetry(response)) {
```

Comment:
I really wanted to add eslint here, but it felt like we'd already sunk a lot of time into getting this right.